### ISSUE TITLE
EIP-1559 transcations unsuitable for Polygon 

### ISSUE THREAD

**Issue by JABirchall** [maticnetwork/bor#384]
#### System information
Im not using a local bor

#### Expected behaviour
Polygon advertises on the website that the network can handle 65,000 tx/s, while in reality the network struggles with 200 tx/s. Gas prices sky rocket when tx/s reaches the hundreds leaving alot of transcations sitting in the mempool.

#### Actual behaviour
Polygon should be able to handle the 65k tx/s as advertised
Ive monitored the network durring high activity and noticed transcations per block seems to topout at 256 tx/per block, resulting in 128 tx/s. Durring these high activity times gas price can increase upto 100x, while a normal gasprice for polygon is 32gwei, when blocks reach 256tx gas price can increase too 1000gwei and higher.

There seems to a hard limit of 256 tx per block somewhere in the polygon network. 

I'd expect the network to handle more transcations as advertised and for gas price to increase minimally

#### Steps to reproduce the behaviour
watch the network durring a high activity. 

#### Backtrace
none

When submitting logs: please submit them as text and not screenshots.

**Comment by JABirchall** [comment]
Today the gas fee on polygon reached 50000 gwei. I was able to deduce the culprate being the london block, specifically EIP-1559. 
Since EIP-1559 fee calcuatiuon is designed for ethereums slower transcations, its is completely unsutable for the polygon network.  

The EIP-1559 block gas limit is too small the polygon chain, which is limiting the networks tx/s, each block has a gas limit of 30 million, and 100 transcations are using this completely. Each time this happens, EIP-1559 gas calcuation increases the gas fee by 32. 

EIP-1559 also comes with a mechanism to solve this problem, which is to publish blocks with 0 transcations, this again is completely unsutable for polygon as these blocks are wasted and 0 matic is rewarded to the validator. This work on ethereum as every block has a base reward of 2 eth, but polygons validators are rewarded with the fees of the containing transcations. 

I propose a list of changes to how EIP-1559 is handled on polygon to solve this issues

1. Increase the block gas limit from 30 million to a higher value
2. [Change this code](https://github.com/maticnetwork/bor/blob/045e5c2c4c373c4fd565afea6ea65812566a6542/consensus/misc/eip1559.go#L62) to not target to use half the gas limit before increasing the gas but maybe 80-90% (parent.GasLimit * 0.8)
3. [Change this code](https://github.com/maticnetwork/bor/blob/097812307915e6de97f3924e3d4b6ffbf08eb62b/tests/bor/helper.go#L119) too actually aim to increase / decline the gas limit on demand. ATM is only targeting the same gas limit every block. even tho a [min and max](https://github.com/maticnetwork/bor/blob/2321e64ed7f14a82776c854607148810f8389d17/params/protocol_params.go#L23-L24) is set.

**Comment by ssandeep** [comment]
Hey Jake, thanks for taking the time and for your proposed solutions. Let me address our take below:
- When Polygon launched, we used an adapted version of Plasma Framework that provided a solution for fast and extremely low-cost transactions, with a theoretical limit of 65,000 TPS. Since then, we’ve pivoted to Geth (Go-Ethereum) – a more used and well-tested client. With Geth, we hit the limit of 7,000 TPS in Devnet.
- It’s important to note that this change hasn’t affected the PoS chain performance and has its own set of benefits, as Geth is interoperable with many existing tools, including wallets, IDEs, and developer tools like Metamask and Remix. In fact, we currently process over 3 million transactions a day and have onboarded over 19,000 Dapps in our Ecosystem.

Regarding the transaction fee mechanics of EIP1559,
- We are aiming for an average of 15M per block. So if demand for block space increases, we increase the base fee. Whether that is 50K gwei or 5 gwei depends on the demand. If we decide to increase the fee, the average gas consumption will also need to be changed.
- From what we’ve seen, the empty blocks occur due to Blockchain bloat and not as a result of EIP1559.
- In general, rewards on Polygon are different from eth. The main rewards are on checkpointing, the miner tip being a bonus.
- We acknowledge that we need to fine tune some params in a way that works best for our chain based on the block time and gas limits on our chain as compared with Ethereum. We have already started working on analysing the data and identifying changes to be done. We will also be testing these changes thoroughly before going live on Mainnet as we don't want cause any disruptions to users/dapps.
